GoogleAnalyticsImporter.php

```diff
@@ -61,12 +63,45 @@ public function getListHooksRegistered()
             'API.Request.dispatch.end' => 'translateNotSetLabels',
             'SitesManager.deleteSite.end'            => 'onSiteDeleted',
             'Template.jsGlobalVariables' => 'addImportedDateRangesForSite',
+            'Rules.isRequestAuthorizedToArchive' => 'isRequestAuthorizedToArchive',
         ];
     }
 
+    public function isRequestAuthorizedToArchive(&$isRequestAuthorizedToArchive, Parameters $params)
+    {
+        if ($params->getPeriod()->getLabel() != 'day') {
```

Comment:
btw @diosmosis are the params always set? Or can they be null?  Just asking because there is a default value of null set ? https://github.com/matomo-org/matomo/pull/15457/files?w=1#diff-52be076826a74ae43ce677c1457f7aa4R229

Reply:
It will only post the event if `$params` isn't null, so should be ok.